server/app/models/stack.rb

```diff
@@ -14,10 +15,34 @@ class Stack
 
   index({ grid_id: 1 })
   index({ name: 1 })
+  index({ parent_name: 1 })
```

Comment:
Needs a migration to actually create the index, it just needs to call `Stack.create_indexes`.